tests/Features/ReturnTypes/ContainerArrayAccessTest.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Features\ReturnTypes;
+
+use Illuminate\Auth\AuthManager;
+use Illuminate\Contracts\Foundation\Application;
+
+class ContainerArrayAccessTest
+{
+    /** @var Application */
+    protected $app;
+
+    public function testAuth(): AuthManager
```

Comment:
Can you also add a test just bellow, that resolves $this->app['fooooooooo']; ? I am curious about the result.

Reply:
It would throw `BindingResolutionException` exception.

Reply:
Don't merge this yet. I want to run some tests.